pyomo/gdp/util.py

```diff
@@ -72,3 +75,27 @@ def clone_without_expression_components(expr, substitute):
                 _idx = 0
                 _len = len(_argList)
     return ans[0]
+
+
+def target_list(x):
```

Comment:
This seems like a useful function even outside of GDP. Maybe move this to general Pyomo utils?

Reply:
Maybe after a little more refinement.  Working through this function highlighted some "challenges" with transformations that I'd like to think through before promulgating this as the recommended solution.